tmt/steps/provision/__init__.py

```diff
@@ -1588,6 +1588,42 @@ def check_boot_time() -> None:
         self.debug("Connection to guest succeeded after reboot.")
         return True
 
+    def reboot(
+            self,
+            hard: bool = False,
+            command: Optional[Union[Command, ShellScript]] = None,
+            timeout: Optional[int] = None,
+            tick: float = tmt.utils.DEFAULT_WAIT_TICK,
+            tick_increase: float = tmt.utils.DEFAULT_WAIT_TICK_INCREASE) -> bool:
+        """
+        Reboot the guest, and wait for the guest to recover.
+
+        :param hard: if set, force the reboot. This may result in a loss of
+            data. The default of ``False`` will attempt a graceful reboot.
+        :param command: a command to run on the guest to trigger the reboot.
+        :param timeout: amount of time in which the guest must become available
+            again.
+        :param tick: how many seconds to wait between two consecutive attempts
+            of contacting the guest.
+        :param tick_increase: a multiplier applied to ``tick`` after every
+            attempt.
+        :returns: ``True`` if the reboot succeeded, ``False`` otherwise.
+        """
+
+        if hard:
+            raise tmt.utils.ProvisionError(
+                "Method does not support hard reboot.")
```

Comment:
Mention the provision method as well?

Reply:
Yeah, but I don't know how to get the actual name :) We're in a guest instance, we can get the guest class name, but there's no link to a plugin. How about using a guest (multihost) name instead? Changed in be4a36bb.

Reply:
Yeah, makes sense, thanks!